pkg/shell/hosts_dialog.jsx

```diff
@@ -200,6 +222,74 @@ class NotSupported extends React.Component {
     }
 }
 
+class Connect extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            inProgress: false,
+        };
+    }
+
+    onConnect() {
+        window.sessionStorage.setItem("connection-warning-shown", true);
+        this.setState({ inProgress: true });
+        this.props.run(try2Connect(this.props.machines_ins, this.props.full_address), ex => {
+            let keep_message = false;
+            if (ex.problem === "no-host") {
+                let host_id_port = this.props.full_address;
+                let port = "22";
+                const port_index = host_id_port.lastIndexOf(":");
+                if (port_index === -1) {
+                    host_id_port = this.props.full_address + ":22";
+                } else {
+                    port = host_id_port.substr(port_index + 1);
+                }
+
+                ex.message = cockpit.format(_("Unable to contact the given host $0. Make sure it has ssh running on port $1, or specify another port in the address."), host_id_port, port);
+                ex.problem = "not-found";
+                keep_message = true;
+            }
+            this.setState({ inProgress: false });
+            this.props.setError(ex, keep_message);
+        });
+    }
+
+    render() {
+        return (
+            <Modal id="hosts_connect_server_dialog" isOpen
+                   position="top" variant="small"
+                   onClose={this.props.onClose}
+                   title={fmt_to_fragments(_("Connect to $0?"), <b>{this.props.host}</b>)}
+                   titleIconVariant="warning"
+                   footer={<>
+                       <HelperText>
+                           <HelperTextItem>{_("You will be reminded once per session.")}</HelperTextItem>
+                       </HelperText>
+                       <Button variant="warning" isLoading={this.state.inProgress}
+                                       onClick={() => this.onConnect()}>
+                           {_("Connect")}
+                       </Button>
+                       <Button variant="link" className="btn-cancel" onClick={this.props.onClose}>
+                           { _("Cancel") }
+                       </Button>
+                   </>}
+            >
+                <TextContent>
+                    <Text component={TextVariants.p}>
+                        {_("Remote hosts have the ability to run arbitrary code on all connected hosts. Only connect to machines that you trust.")}
```

Comment:
I really don't like the "arbitrary code" as it is an almost meaningless phrase.

Perhaps something like this:

```suggestion
                        {_("All connected hosts are able to control each other. This may include running programs, accessing files, and even executing possibly harmful scripts or exploits. Only connect to machines you fully trust.")}
```

It's a bit more explicit with details. WDYT? Do you have any suggestions that build on top of this? Or others that would explain it?

Reply:
Thanks, this is nice! Possibly trimming it a bit?

> All connected hosts are able to control each other. This may include accessing files and running possibly harmful programs. Only connect to machines you fully trust.

Reply:
Drop the "may"?

Reply:
> All connected hosts are able to control each other. This includes accessing files and running harmful programs. Only connect to machines you fully trust.

Reply:
"Connected hosts can control each other. This may include actions such as running harmful programs and accessing sensitive files. Only connect to machines you fully trust."

Reply:
"Connected hosts can fully control each other. This includes running programs that could harm your system or steal data. Only connect to trusted machines."